sympy/stats/crv_types.py

```diff
@@ -2897,6 +2997,11 @@ def pdf(self, x):
         alpha, beta = self.alpha, self.beta
         return beta * (x/alpha)**(beta - 1) * exp(-(x/alpha)**beta) / alpha
 
+    def _moment_generating_function(self, t):
+        lamda, k = self.alpha, self.beta
+        n = Dummy('n')
+        return Sum(t**n * lamda**n  * gamma(1 + n/k) / factorial(n), [n, 0, oo])
```

Comment:
Can this be used for computing moments? Can you add examples?

Reply:
I guess not since the series expansion of a sum will fail. It's possible there's a hypergeometric representation of the MGF for the Weibull distribution but I haven't seen written out anywhere.

Reply:
I was suspecting that the sum would remain unevaluated. Perhaps this should also raise `NotImplementedError` (or return `None`).

Reply:
> with the integral formula for some specific cases

Can the sum be replaced by an integral?

Reply:
The integral is already used- if the _moment_generating_function method is not overridden in the child class it will be computed.

Reply:
That is was my understanding. So if this method cannot be used for computing moments in general, then would it be better to not override the _moment_generating_function in this class?

Reply:
That's my intent, yes.